### DISCUSSION TITLE
Kong Gateway token

### DISCUSSION THREAD

**Discussion by chermeux** [Kong/kong#13407] [Help]
I'm doing a microservice architecture project. So I have several microservices and a Kong gateway api. My microservices are made with django and they all have jwt tokens. Thus, we can only access the data of each microservice if we have a token and a refresh token. Each microservice has an administrator user, named here admin and with a password adminpassword (I specify this is a random example). For example, with postman, I would like to be able to access the data of a microservice without having to connect to it. I actually want the API gateway to take care of the connection with the microservices. In a way, I'm removing the security of each microservice, but it's for a more global project. Thus, api gateway would always have the information to connect to each microservice and renew the tokens. And postman would allow me to access the data of each microservice if I go through the gateway api. I can't figure out how to do that. Can you help me, that would be very nice.

I've already connected all my microservices but I don't see how to get Kong to manage the tokens himself with the microservices.

**Comment by xianghai2** [comment]
You may look into the function: https://docs.konghq.com/hub/kong-inc/pre-function/
Also request transformer: https://docs.konghq.com/hub/kong-inc/request-transformer/
And check if this could solve the problem. thanks.
